initfiles/bin/init_backupnode.in

```diff
@@ -0,0 +1,111 @@
+    log "starting backupnode job..."
+
+
+    for target in ${thorlist[@]}; do
+        log "performing backup on target cluster : ${target}"
+        cfg.section.${target}
+        backupnode_data=$(@EXEC_PATH@/updtdalienv @CONFIG_DIR@/@ENV_XML_FILE@ -d data backupnode ${target})
+        if [[ -z "$backupnode_data" ]]; then
+            log "cannot determine backupnode directory for ${target}"
+            break
+        fi
+
+        // fetch thorgroup slave machines from dali
+        @EXEC_PATH@/daliadmin server=$daliserver dfsgroup ${target} @RUNTIME_PATH@/${target}/backupnode.slaves
+        if [[ $? != 0 ]]; then
+            log "failed to lookup dali group for $target"
+            break
+        fi
+
+        backupnode_data=$backupnode_data/${target}/last_backup
+        if [[ "$localthor" == "true" ]]; then
+            backupnode_remotedata=$backupnode_data
+        else
+            backupnode_remotedata=//${thormaster}$backupnode_data
+        fi
+
+        mkdir -p $backupnode_data
+        rm -f $backupnode_data/*.ERR
+        rm -f $backupnode_data/*.DAT
+
+        log "Using backupnode directory $backupnode_data"
+        log "Reading slaves file @RUNTIME_PATH@/${target}/backupnode.slaves"
+        log "Scanning files from dali ..."
+
+        nodegroup=$thorprimary
+        if [[ -z "$thorprimary" ]]; then
+            nodegroup=$thorname
+        fi
+
+        @EXEC_PATH@/backupnode -O $daliserver $nodegroup $backupnode_data >> $logfile 2>&1
+        if [[ $? -ne 0 ]]; then
+            log "Backupnode of ${target} failed - see ${logfile}"
+            break
+        fi
+
+        # maximum number of threads frunssh will be permitted to use (capped by # slaves)
+        MAXTHREADS=1000
+
+        frunssh @RUNTIME_PATH@/${target}/backupnode.slaves "killall backupnode 1>/dev/null 2>&1" -i:$SSHidentityfile -u:$SSHusername -pe:$SSHpassword -t:$SSHtimeout -a:$SSHretries -n:$MAXTHREADS -b >> $logfile 2>&1
+        log "frunssh @RUNTIME_PATH@/${target}/backupnode.slaves \"/bin/sh -c 'mkdir -p @LOG_PATH@; mkdir -p @RUNTIME_PATH@; @EXEC_PATH@/backupnode -T -X $backupnode_remotedata %n %c %a %x $2 > @LOG_PATH@/backup_node_%n_${logpthtail}.log 2>&1'\" -i:$SSHidentityfile -u:$SSHusername -pe:$SSHpassword -t:$SSHtimeout -a:$SSHretries -n:$MAXTHREADS -b >> $logfile 2>&1"
+        frunssh @RUNTIME_PATH@/${target}/backupnode.slaves "/bin/sh -c 'mkdir -p @LOG_PATH@; mkdir -p @RUNTIME_PATH@; @EXEC_PATH@/backupnode -T -X $backupnode_remotedata %n %c %a %x $2 > @LOG_PATH@/backup_node_%n_${logpthtail}.log 2>&1'" -i:$SSHidentityfile -u:$SSHusername -pe:$SSHpassword -t:$SSHtimeout -a:$SSHretries -n:$MAXTHREADS -b >> $logfile 2>&1
+
+        log "------------------------------"
+        log "Waiting for backup to complete"
+
+        log "nohup backupnode -W @RUNTIME_PATH@/${target}/backupnode.slaves $backupnode_data >> $logfile 2>&1 &"
+        nohup backupnode -W @RUNTIME_PATH@/${target}/backupnode.slaves $backupnode_data >> $logfile 2>&1
```

Comment:
Yes it does need to run in the background.  Had it in the foreground for testing.  Good catch.